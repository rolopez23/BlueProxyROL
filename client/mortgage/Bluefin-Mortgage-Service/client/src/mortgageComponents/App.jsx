/* eslint-disable import/extensions */
import React from 'react';
import Styled from 'styled-components';
import axios from 'axios';
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';
// Component Imports
import AdList from './ads/AdList.jsx';
import PaymentCalculator from './paymentCalculator/PaymentCalculator.jsx';
// Model imports
import Mortgage from '../../mortgageModel/mortgage.js';

const AppWrapper = Styled.div`
  border-top: 1px solid #e2e2e2;
  display: block;
  position: relative;
  margin-top: 3rem;
  min-width: 600px;
  width: 728px;
  maxwidth: 728px;
`;


class App extends React.Component {
  constructor(props) {
    super(props);
    const { listingId } = this.props;
    this.state = {
      listingId,
      price: null,
      downPayment: null,
      rate: 3.5,
      ads: [],
      listing: {},
      payments: [],
      loanType: '30 Year Fixed',
      years: 30,
    };
    this.updatePrice = this.updatePrice.bind(this);
    this.updateMortgage = this.updateMortgage.bind(this);
    this.updateRate = this.updateRate.bind(this);
    this.updateLoanType = this.updateLoanType.bind(this);
  }

  componentDidMount() {
    const { listingId } = this.state;
    const url = `/listing${listingId}`;
    axios.get(url)
      .then((response) => {
        // console.log(response.data);
        const newAds = response.data.ads;
        const newListing = response.data.listing;
        const { price } = newListing;
        const downPayment = price * 0.2;
        this.updateMortgage(30, 3.5, price, downPayment);
        this.setState({
          ads: newAds,
          listing: newListing,
          price,
          downPayment,
        });
      });
  }

  // Section for helpers that update main App state in lower components

  // returns a payment train, creates a mortgage object but only saves payment
  // array to state
  // Inputs: Required: years, APR, price
  // Optional: downPayment, HOA, taxRate, homeownersInsurance, PMI
  updateMortgage(years, APR, price, downPayment, HOA, taxRate, homeownersInsurance, PMI) {
    const mortgage = new Mortgage(years, APR, price, downPayment, HOA, taxRate, homeownersInsurance, PMI);
    const payments = mortgage.getMonthlyPayment();
    // console.log(payments);
    this.setState({ payments });
    // console.log(this.state.payments);
  }

  updateLoanType(loanType) {
    let newYears = 30;
    const { rate, price, downPayment } = this.state;
    if (loanType === '15 Year Fixed') {
      newYears = 15;
    }
    this.setState({ loanType, years: newYears });
    this.updateMortgage(newYears, rate, price, downPayment);
  }

  updatePrice(price, downPayment) {
    const { rate, years } = this.state;
    this.setState({
      price,
      downPayment,
    });
    this.updateMortgage(years, rate, price, downPayment);
    // console.log('updated price', price, downPayment);
  }

  updateRate(rate) {
    const { years, price, downPayment } = this.state;
    this.setState({ rate });
    this.updateMortgage(years, rate, price, downPayment);
  }

  render() {
    const {
      listing, ads, price, downPayment, payments, loanType, rate,
    } = this.state;
    // console.log(price, downPayment);
    return (
      <AppWrapper>
        <PaymentCalculator
          rate={rate}
          payments={payments}
          loanType={loanType}
          price={price}
          downPayment={downPayment}
          updatePrice={this.updatePrice}
        />
        <AdList
          ads={ads}
          listing={listing}
          rateChange={this.updateRate}
          loanTypeChange={this.updateLoanType}
        />
      </AppWrapper>
    );
  }
}

App.propTypes = {
  listingId: PropTypes.number.isRequired,
};


export default App;
