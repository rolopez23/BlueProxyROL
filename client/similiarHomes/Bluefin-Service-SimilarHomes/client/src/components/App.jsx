import React from 'react';
import axios from 'axios';
import styled from 'styled-components';

// eslint-disable-next-line import/extensions
import SimilarListingsContainer from './SimilarListingsContainer.jsx';

class NearbyListingsContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      listings: [],
    };
  }

  componentDidMount() {
    const { listings } = this.state;
    if (!{ listings }.length) {
      this.getListings();
    }
  }

  getListings() {
    axios.get('http://localhost:4004/image')
      .then((listings) => {
        this.setState({
          listings: listings.data,
        });
      });
  }

  render() {
    const { listings } = this.state;
    const containers = listings.map((l) => <SimilarListingsContainer data={l} />);

    const ScrollingComponent = styled.div`
      display: grid;
      grid-gap: 16px;
      grid-template-columns: repeat(${listings.length}, 297px);
      grid-template-rows: 1fr;
      width: 616px;
      overflow-y: none;
      overflow-x: scroll;
      scroll-snap-type: inline mandatory;
    `;

    const ScrollingContainer = styled.div`
      height: 400px;
    `;

    const output = (
      <div id="section-title">
        <h2 style={{ 'font-family': '"Libre Franklin", sans-serif' }}>Nearby Similar Homes</h2>
        <ScrollingContainer>
          <ScrollingComponent className="similar-listings">
            {containers}
          </ScrollingComponent>
        </ScrollingContainer>
      </div>
    );
    return output;
  }
}

export default NearbyListingsContainer;
