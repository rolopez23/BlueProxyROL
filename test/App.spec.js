import React from 'react';
import renderer from 'react-test-renderer';
import App from '../client/src/components/App.jsx';


describe('My Test Suite', () => {
  it('My Test Case', () => {
    expect(true).toEqual(true);
  });
});


describe('App', () => {
  test('App exists', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});