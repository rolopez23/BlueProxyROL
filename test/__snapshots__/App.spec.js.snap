// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App exists 1`] = `
<div>
  I'm an App!
</div>
`;
